source/AI.cpp

```diff
@@ -455,7 +455,8 @@ void AI::UpdateEvents(const list<ShipEvent> &events)
 				// If you provoke the same ship twice, it should have an effect both times.
 				if(event.Type() & ShipEvent::PROVOKE)
 					newActions |= ShipEvent::PROVOKE;
-				event.TargetGovernment()->Offend(newActions, target->RequiredCrew());
+
+				event.TargetGovernment()->Offend(newActions, target->Crew(), target->Cost());
```

Comment:
I feel like it'd be simpler to allow ships to have a "fake crew" attribute of some sort. Ships could then be given this "fake crew" attribute, and it would add to the reputation hit for interacting with a certain ship without tying reputation to ship cost. Said ship then wouldn't even necessarily need to be a drone; perhaps a faction has crewed ships of a certain type that it takes more offense at having be destroyed compared to other ships. You then wouldn't need separate penalties just for drones and you could cut down on making any changes to Government and Politics (excluding the addition of the outfit scan penalty, that is), only needing to add a new function to Ship to grab this "fake crew" value.

For example the line here would become:
```
				event.TargetGovernment()->Offend(newActions, target->Crew() + target->FakeCrew());
```
And in Ship you might add something like:
```
int Ship::FakeCrew() const
{
	return attributes.Get("fake crew");
}
```
Or even just change the line here to:
```
				event.TargetGovernment()->Offend(newActions, target->Crew() + target->Attributes().Get("fake crew"));
```
and then you wouldn't even need to worry about adding new functions to Ship.
Although I imagine we can think of a better attribute name than "fake crew".

Reply:
Sure, I would be good with this approach too 

"Crew equivalent" maybe? Or "Life value"

As an expansion of this thought, what if it was more of an equivalent cost that got added to the normal crew cost? So if a ship has a crew equivalent of ten, and has twenty crew, then it's impact would be like thirty crew. Although this does reveal the problem that it is so specific rather than government specific.

Reply:
I don't think that the cost of ships should directly factor into reputation changes. With some "fake crew" attribute or similar, a content creator could allow the cost of a ship to factor into the reputation change by having the "fake crew" scale with the ship cost, but it wouldn't be forced.

Reply:
I think I should do a method LifeValue() for a ship that returns the crew + life value and we can easily change such function. Default life value will be 0 anyway so it doesn't matter if we add it to crew (or requiredCrew?).
Idk how to make such a change government specific tough, maybe have a LifeValuePer500k parameter that can be added in government.txt and just be used for drones ? Thing is most governments only have one drone ship anyway.
Removing the specific penalties for drones means less flexibility tough but I'm okay with it.

Reply:
OK, so stepping back a bit to summarize here. 

I think we  basically need two things here:
1. Something that determines the reputational value of the ship hull itself.
2. Something that specifies how much a particular government cares about the hull value. 

Simplest implementation I can think of comes down to two additional values:
A. Inside a ship definition, have a `Crew Equivalent` value that basically translates to how complex and advanced a ship is. The default value is 0
B. Inside the government definition, have a `Hull appreciation` value that determines how much the government actually cares about the `Crew Equivalent` value.

Potential Usage: 
When determining the reputational effect of killing a shp, the formula wold expand from
crew * penalty_value = final_penalty
to
(crew + (Crew_Equivalent * Hull_Appreciation)) * penalty_value = final_penalty

How does that sound? That way both the ship and the government have a bearing on what the reputation effect is, and it allows for quite a bit of control. It's not quite as fine grained as my original idea of having a drone specific set of penalties, but I think I'm fine with that.

Reply:
> So there should be a LifeValue() method for ships that returns the crew + life value and we can easily change such function. Default life value will be 0 anyway so it doesn't matter if we add it to crew (or requiredCrew if > than crew in case of capturing).



> Idk how to make such a change government specific tough
> Removing the specific penalties for drones means less flexibility tough but I'm okay with it.

No, I'm saying to *not* make the change government specific. If all we have is some sort of "fake crew" attribute, then that allows us to increase the reputational hit for destroying *any* ship, not just drones. Then if a content creator wants to make a specific government care about the destruction of their drones, they'd simply add this "fake crew" attribute to the drone that that government uses. Then we don't need to worry about touching anything in Government or Politics, and we don't need to worry about having some scale between the cost of a ship and its reputational hit, because that scale would be defined by the content creator.

This is actually a *more* flexible system, not less.

Reply:
okay I'm just wandering you say Attributes.Get("fake crew") I guess I gotta use that function on the ship but why not just make a getter for crew equivalent like done for crew?

Reply:
> okay I'm just wandering you say Attributes.Get("fake crew") I guess I gotta use that function on the ship but why not just make a getter for crew equivalent like done for crew?

Because "crew" isn't an attribute defined by the ship. The ship defines the total number of bunks and the required crew level as attributes, but the current crew level is not itself an attribute.

> I could do even better (in my opinion) and make a getter for max(crew, required crew) + crew equivalent

That would have almost no effect on the Offend called in AI (as crew is above required crew for >99% of situations for NPC ships, the only exception being you begin a capture attempt then later bail out) and would never have an effect on the Offend called in BoardingPanel (as the crew must necessarily be 0 before that Offend is called).